File: backend/src/main/java/org/example/backend/domain/aquarium/dto/AquariumResponseDto.java
package org.example.backend.domain.aquarium.dto;

import java.time.LocalDateTime;
import org.example.backend.domain.aquarium.entity.Aquarium;

public record AquariumResponseDto(
    Long aquariumId,
    String aquariumName,
    LocalDateTime createDate,
    int notifyCycleDate,
    LocalDateTime lastNotifyDate,
    LocalDateTime nextNotifyDate
) {
  public AquariumResponseDto(Aquarium aquarium) {
    this (
        aquarium.getId(),
        aquarium.getName(),
        aquarium.getCreateDate(),
        aquarium.getCycleDate(),
        aquarium.getLastDate(),
        aquarium.getNextDate()
    );
  }
}

File: backend/src/main/java/org/example/backend/domain/aquarium/dto/AquariumResponseDto.kt
package org.example.backend.domain.aquarium.dto

import org.example.backend.domain.aquarium.entity.Aquarium
import java.time.LocalDateTime

data class AquariumResponseDto(
    val aquariumId: Long,
    val aquariumName: String,
    val createDate: LocalDateTime,
    val notifyCycleDate: Int,
    val lastNotifyDate: LocalDateTime?,
    val nextNotifyDate: LocalDateTime?
) {
    constructor(aquarium: Aquarium) : this(
        aquarium.id,
        aquarium.name,
        aquarium.createDate,
        aquarium.cycleDate,
        aquarium.lastDate,
        aquarium.nextDate
    )
}
